Aura CLI usage guide (#622)
neo4j/docs-aura: modules/ROOT/pages/aura-cli/migration.adoc

= Migrating to the Aura CLI
:description: Migrate to the Neo4j Aura command line interface.

> Aura CLI  has evolved from a Neo4j Labs to a supported Neo4j product.

Author: do you suggest linking to the labs page/project?

> Aura CLI  has evolved from a Neo4j Labs to a supported Neo4j product.

Author: added a link to the labs page

= Migrating to the Aura CLI
:description: Migrate to the Neo4j Aura command line interface.

> Aura CLI  has evolved from a Neo4j Labs to a proper Neo4j product.

Reviewer: ```suggestion
Aura CLI  has evolved from a Neo4j Labs to a supported Neo4j product.
```

The Neo4j Labs Aura CLI will continue to be available for installation for the foreseeable future, but without any further development.

> We recommend you to move to the Aura CLI as soon as you are able.

Reviewer: ```suggestion
It is recommended that you move to the Aura CLI as soon as you are able.
```

This section outlines what you must consider to move the Aura CLI.

== Behavior

> * The name of the CLI is now "aura-cli".

Reviewer: ```suggestion
* The name of the CLI is now `aura-cli`.
```

* Support for using environmental variables has been removed as they can be visible in process listings and can be accidentally logged making them vulnerable to exposure.
  For sensitive values e.g. Aura API client ID and client secret, a secrets manager is recommended.
* Neo4j Labs Aura CLI used plural names for commands; the new Aura CLI has singular naming instead.
* It is not possible to return the raw API response body with the Aura CLI.
* Flags with the Aura CLI do not have a short form and must be specified in full length.

== Feature set

At the time of general availability, the Aura CLI supports these new features compared to the previous Labs project:

> * Customer-managed keys

Reviewer: ```suggestion
* Customer managed keys
```

Author: above, customer managed keys was capitalized - should it be capitalized here as well?

Reviewer 2: Yes please (it's no biggie though, users don't mind so much about the capitalization)

* Beta of GraphQL Data APIs

== Installation

Download the Aura CLI from link:https://github.com/neo4j/aura-cli/releases[https://github.com/neo4j/aura-cli/releases].
It is available as a native binary for Windows, Linux and Mac platforms.

> After downloading the relevant file and extracting the executable, the Aura CLI can be run immediately; there are no dependencies.

Reviewer: ```suggestion
After downloading the relevant file and extracting the executable, the Aura CLI can be run immediately as there are no dependencies.
```

== Commands

[cols="3,^1,3", options="header"]
|====
| Labs Aura CLI                      | maps to | Aura CLI
| config                             | →       | config
| credentials                        | →       | credential       
| instances                          | →       | instance         
| snapshots                          | →       | instance → snapshot               
| tenants                            | →       | tenant            
| tenants → get-metrics-integration  | →       | instance get `<InstanceID>` returns the URL for metrics integration
|                                    |         | **NEW**: customer-managed-key
|                                    |         | **NEW**: data-api (beta)
|====

== Command and flag changes with Aura CLI

As far as was possible, the Aura CLI commands and their flags are the same as for the Labs based Aura CLI.

> However, there are some changes.
> See the following sections.

Reviewer: ```suggestion
However, there are some changes as outlined in the following sections.
```

== Config

It is not possible to remove a setting and its value although the value itself can be changed.

> If deletion is wanted, then remove it from the JSON configuration file, either _$HOME/Library/Preferences/neo4j/cli/config.json_ (MAC) or _%LOCALAPPDATA%\neo4j\cli\config.json_ (Windows).

Reviewer: Shouldn't these be in a code block?

Author: what's our convention for file paths?

Author: it's now in backticks

== Create a new AuraDB

When creating an Aura instance with the Aura CLI, the memory size is specified with an integer appended with "GB" e.g. "8GB".
Previously, this was just the integer.

== Delete an AuraDB

The Aura CLI does not prompt for confirmation when deleting an AuraDB and starts the operation immediately.

== Update an AuraDB

Only the ID can be used to identify the AuraDB to update.
The same properties can be changed as previously: its name and memory size.

== Get AuraDB instance details

This now includes the URL to obtain metrics.

> == Tenants

Reviewer: Again with tenants...
